### PR TITLE
Add and update contracts based on the new interface

### PR DESCRIPTION
**Pull request by feeblefakie**
Added sample contracts based on the new I/F.
Adding sample functions will be done in another PR.

### CODE REVIEW

**Review by supl — APPROVED** [review]
LGTM.
Thank you!

I have a curious question.
What is the consideration about using *StateUpdaterReader?
Could *StateUpdater return the updated value?

**Comment by feeblefakie** [comment]
@supl Sorry for the bad name. It's a mix of StateUpdater and StateReader to test nested invocation.
It updates an asset using StateUpdater and reads the asset using StateReader.

File: src/main/java/com/org1/contract/JacksonBasedStateUpdater.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import com.scalar.dl.ledger.exception.ContractContextException;
+import com.scalar.dl.ledger.statemachine.Asset;
+import com.scalar.dl.ledger.statemachine.Ledger;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+/**
+ * The contents of the contract are the same as StateUpdater. It is recommended to use
+ * JacksonBasedContract for taking a good balance between development productivity and performance.
+ */
```
Comment (jnmt) [inline src/main/java/com/org1/contract/JacksonBasedStateUpdater.java]: Should we add this kind of message for `JacksonBasedStateReader` and `JacksonBasedStateUpdaterReader` as well?

**Review by jnmt — APPROVED** [review]
LGTM! Thank you! I left a very minor question, which does not block merging.

File: src/main/java/com/org1/contract/JacksonBasedStateUpdater.java (hunk 1 above)
Comment (feeblefakie) [reply src/main/java/com/org1/contract/JacksonBasedStateUpdater.java]: Thank you for the suggestion.
Updated in [396a068](https://github.com/scalar-labs/scalardl-java-client-sdk/pull/28/commits/396a0689b55388fd38e743ff5f2a69e763ba6b78).
The commit also added the links to the sources.

**Pull request merged**